cpp/src/arrow/status.cc

```diff
@@ -131,6 +131,13 @@ void Status::Abort(const std::string& message) const {
   std::abort();
 }
 
+void Status::Warn() const { Warn(std::string()); }
+
+void Status::Warn(const std::string& message) const {
+  ARROW_LOG(WARNING) << "Arrow Warning -- "
```

Comment:
I don't think you need to add the "Arrow Warning" phrase. Instead that should be handled by `ARROW_LOG` itself, or whatever it redirects to.